sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -413,6 +413,12 @@ template <class To, class From> inline To cast(From value) {
   return (To)(value);
 }
 
+// Cast for std::vector<cl_event>, according to the spec, make_event
+// should create one(?) event from a vector of cl_event
+template <class To> inline To cast(std::vector<cl_event> value) {
+  return (To)(value[0]);
```

Comment:
Please correct me if I am wrong.
It does not look like a correct interpretation of the vector of events. In spec (Table 185): 
```
A SYCL event can encapsulate one or multiple OpenCL events, representing a number of dependencies in the same or different contexts, that must be satisfied for the SYCL event to be complete.
```
If you use the first event as a representation of th evector only it will be waited for the SYCL event to be complete.

Reply:
This is temporary workaround as the SYCL2020 spec may contain a bug that should be [discussed](https://gitlab.khronos.org/sycl/Specification/-/issues/588).
In addition, there are more misunderstandings with the specification:
1) At this moment we don't have sycl::event [constructors](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#table.constructors.event) which takes vector as a parameter. 
2) From event class [description](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:interface.event): "Note that, although an event represents the status of a **particular operation**, the dependencies of a certain event can be used to keep track of multiple steps required to synchronize said operation." whereas from the [table 185](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#table.opencl.interop) "A SYCL event can encapsulate one or multiple OpenCL events, representing a number of dependencies in the same or different contexts, that must be satisfied for the SYCL event to be complete."